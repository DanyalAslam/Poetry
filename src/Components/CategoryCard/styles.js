import {StyleSheet} from 'react-native';
import {vw, vh} from '../../Units';
import {appTheme} from '../../Utils';
import fonts from '../../assets/fonts';

export default Styles = StyleSheet.create({
  ripple: {
    height: 18 * vh,
    width: '100%',
    marginVertical: 1 * vh,
    elevation: 4,
    backgroundColor: 'white',
    borderRadius: 3 * vw,
    shadowColor: 'black',
  },
  Container: {
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3 * vw,
  },
  title: {
    color: appTheme.white,
    fontSize: 3 * vh,
    fontFamily: fonts.SSR,
  },
});
