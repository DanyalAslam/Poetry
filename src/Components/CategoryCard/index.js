import React, { Component } from 'react'
import { Text, ImageBackground } from 'react-native'
import styles from './styles.js'
import RippleTouch from '../RippleTouch/index.js'


const CategoryCard = (props) => {


    return (<RippleTouch style={styles.ripple} rippleColor="black" onPress={props.onPress}>
        <ImageBackground
            style={styles.Container}
            imageStyle={styles.imageStyle}
            source={props.source}
        >
            <Text style={styles.title}>
                {props.title}
            </Text>

        </ImageBackground>
    </RippleTouch>
    )
}

export default CategoryCard