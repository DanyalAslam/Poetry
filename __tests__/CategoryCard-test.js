/**
 * @format
 */

import 'react-native';
import React from 'react';
import CategoryCard from '../src/Components/CategoryCard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<CategoryCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
 