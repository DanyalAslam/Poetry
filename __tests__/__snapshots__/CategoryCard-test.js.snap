// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 22.5,
      "elevation": 4,
      "height": 240.12,
      "marginVertical": 13.34,
      "shadowColor": "black",
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      progressiveRenderingEnabled={true}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          Object {
            "borderRadius": 22.5,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "SourceSerifPro-Regular",
          "fontSize": 53.36,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  />
</View>
`;
